read-fonts/src/tables/postscript/string.rs

```diff
@@ -0,0 +1,504 @@
+//! PostScript string identifiers.
+
+/// PostScript string identifier (SID).
+#[derive(Copy, Clone, PartialEq, Eq, Hash, Debug)]
+pub struct StringId(u16);
+
+impl StringId {
+    pub fn new(raw: u16) -> Self {
+        Self(raw)
+    }
+
+    /// Resolves the identifier as a standard string.
+    ///
+    /// If the identifer represents a standard string, returns `Ok(string)`,
+    /// otherwise returns `Err(index)` with the index that should be used to
+    /// retrieve the string from the CFF string INDEX.
+    ///
+    /// The standard string set is available in the section
+    /// "Appendix A - Standard Strings" at <https://adobe-type-tools.github.io/font-tech-notes/pdfs/5176.CFF.pdf>.
+    pub fn standard_string(self) -> Result<Latin1String<'static>, usize> {
+        let ix = self.0 as usize;
+        if let Some(string) = STANDARD_STRINGS.get(ix) {
+            // The standard strings are all ASCII so it's safe to intepret them
+            // as Latin-1. This is verified in a unit test.
+            Ok(Latin1String::new(string.as_bytes()))
+        } else {
+            Err(ix - STANDARD_STRINGS.len())
+        }
+    }
+}
+
+impl From<i32> for StringId {
+    fn from(value: i32) -> Self {
+        Self::new(value as u16)
+    }
+}
+
+/// Reference to a Latin-1 encoded string.
+///
+/// Strings stored in all PostScript defined fonts are usually ASCII but are
+/// technically encoded in Latin-1. This type wraps the raw string data to
+/// prevent attempts to decode as UTF-8.
+///
+/// This implements `PartialEq<&str>` to support easy comparison with UTF-8
+/// strings.
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+pub struct Latin1String<'a> {
+    chars: &'a [u8],
+}
+
+impl<'a> Latin1String<'a> {
+    pub(crate) fn new(chars: &'a [u8]) -> Self {
+        Self { chars }
+    }
+
+    /// Returns an iterator over the characters of the string.
+    ///
+    /// This simply converts each byte to `char`.
+    pub fn chars(&self) -> impl Iterator<Item = char> + Clone + 'a {
+        self.chars.iter().map(|b| *b as char)
+    }
+}
+
+impl PartialEq<&str> for Latin1String<'_> {
+    fn eq(&self, other: &&str) -> bool {
+        self.chars().eq(other.chars())
+    }
+}
+
+impl std::fmt::Display for Latin1String<'_> {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        for ch in self.chars() {
+            write!(f, "{}", ch)?;
+        }
+        Ok(())
+    }
+}
+
+/// See "Appendix A - Standard Strings" in <https://adobe-type-tools.github.io/font-tech-notes/pdfs/5176.CFF.pdf>
+const STANDARD_STRINGS: &[&str] = &[
```

Comment:
does this want to be `static` instead of `const`? Does using `const` possible duplicate the storage of the strings?


Also sort of an aside but it would be nice to have this declared in one place in the ecosystem, since we also need it during compilation.

Reply:
The str elements are really pointers to separate storage so LLVM will dedup them independently regardless of how this is declared.

Good call on making this available to write-fonts. I’ll make it pub and re-export it.